### PR TITLE
Edit Datasource snippet

### PR DESCRIPTION
**Pull request by xorye**
Fixes https://github.com/redhat-developer/vscode-quarkus/issues/227

To test the Container Image snippet, please try it with a Quarkus project using a recent version of the Qubernetes extension.

image

Signed-off-by: David Kwon <dakwon@redhat.com>

### CODE REVIEW

File: microprofile.ls/com.redhat.microprofile.ls/src/main/resources/com/redhat/quarkus/snippets/quarkus-properties.json [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
-			"quarkus.datasource.driver=${2|org.mariadb.jdbc.Driver,org.mysql.jdbc.Driver,org.h2.Driver,org.postgresql.Driver,com.microsoft.sqlserver.jdbc.SQLServerDriver|}",
+			"quarkus.datasource.url=${1:jdbc:mariadb://localhost:3306/mydb}",
+			"quarkus.datasource.db-kind=${2|h2,derby,mariadb,mssql,mysql,postgresql|}",
 			"quarkus.datasource.username=${3:developer}",
 			"quarkus.datasource.password=${4:developer}",
-			"quarkus.hibernate-orm.database.generation=${5|update,drop-and-create,create,drop,none|}"
+			"quarkus.hibernate-orm.database.generation=${5|none,drop-and-create,create,drop,update|}"
 		],
-		"description": "Configure Quarkus datasource",
+		"description": "Configure Quarkus Panache",
```
Comment (lordofthejars) [inline microprofile.ls/com.redhat.microprofile.ls/src/main/resources/com/redhat/quarkus/snippets/quarkus-properties.json:11]: It is a datasource not Panache. It was good before the change. Even you can use agroal alone without Panache and still need to set all these configuration parameters.

Comment (xorye) [reply microprofile.ls/com.redhat.microprofile.ls/src/main/resources/com/redhat/quarkus/snippets/quarkus-properties.json]: Thank you, fixed

File: microprofile.ls/com.redhat.microprofile.ls/src/main/resources/com/redhat/quarkus/snippets/quarkus-properties.json [hunk 2]
```diff
@@ -2,11 +2,11 @@
 	"Add datasource properties": {
 		"prefix": "qds",
 		"body": [
-			"quarkus.datasource.url=${1|jdbc:mariadb://localhost:3306/mydb,jdbc:mysql://localhost:3306/test,jdbc:h2:mem:mydb,jdbc:postgresql://localhost/mydb,jdbc:sqlserver://localhost:1433;databaseName=mydb|}",
```
Comment (xorye) [inline microprofile.ls/com.redhat.microprofile.ls/src/main/resources/com/redhat/quarkus/snippets/quarkus-properties.json:5]: Double checking, does:
`,jdbc:mysql://localhost:3306/test,jdbc:h2:mem:mydb,jdbc:postgresql://localhost/mydb,jdbc:sqlserver://localhost:1433;databaseName=mydb|`

need to be removed?

**Comment by xorye** [comment]
In my latest push, the `quarkus.datasource.url` snippet completion options remain as before

**Comment by xorye** [comment]
@fbricon @angelozerr  may I merge this?

File: microprofile.ls/com.redhat.microprofile.ls/src/main/resources/com/redhat/quarkus/snippets/quarkus-properties.json [hunk 3]
```diff
@@ -3,10 +3,10 @@
 		"prefix": "qds",
 		"body": [
 			"quarkus.datasource.url=${1|jdbc:mariadb://localhost:3306/mydb,jdbc:mysql://localhost:3306/test,jdbc:h2:mem:mydb,jdbc:postgresql://localhost/mydb,jdbc:sqlserver://localhost:1433;databaseName=mydb|}",
-			"quarkus.datasource.driver=${2|org.mariadb.jdbc.Driver,org.mysql.jdbc.Driver,org.h2.Driver,org.postgresql.Driver,com.microsoft.sqlserver.jdbc.SQLServerDriver|}",
+			"quarkus.datasource.db-kind=${2|h2,derby,mariadb,mssql,mysql,postgresql|}",
```
Comment (fbricon) [inline microprofile.ls/com.redhat.microprofile.ls/src/main/resources/com/redhat/quarkus/snippets/quarkus-properties.json:6]: keep the same order as datasource urls

File: microprofile.ls/com.redhat.microprofile.ls/src/main/resources/com/redhat/quarkus/snippets/quarkus-properties.json [hunk 4]
```diff
@@ -25,5 +25,19 @@
 		"context": {
 			"dependency": "quarkus-jaeger"
 		}
+	},
+	"Add Container Image properties": {
+		"prefix": "cip",
+		"body": [
+			"quarkus.container-image.group=$1",
+			"quarkus.container-image.registry=${2:quay.io}",
```
Comment (fbricon) [inline microprofile.ls/com.redhat.microprofile.ls/src/main/resources/com/redhat/quarkus/snippets/quarkus-properties.json:33]: ${2:|docker.io,quay.io||}

**Comment by xorye** [comment]
fixed

File: microprofile.ls/com.redhat.microprofile.ls/src/main/resources/com/redhat/quarkus/snippets/quarkus-properties.json [hunk 5]
```diff
@@ -3,10 +3,10 @@
 		"prefix": "qds",
 		"body": [
 			"quarkus.datasource.url=${1|jdbc:mariadb://localhost:3306/mydb,jdbc:mysql://localhost:3306/test,jdbc:h2:mem:mydb,jdbc:postgresql://localhost/mydb,jdbc:sqlserver://localhost:1433;databaseName=mydb|}",
```
Comment (fbricon) [inline microprofile.ls/com.redhat.microprofile.ls/src/main/resources/com/redhat/quarkus/snippets/quarkus-properties.json:5]: looks like it's missing a derby url, like "jdbc:derby: mydb". See https://builds.apache.org/job/Derby-docs/lastSuccessfulBuild/artifact/trunk/out/getstart/rgsquck30197.html

Comment (xorye) [reply microprofile.ls/com.redhat.microprofile.ls/src/main/resources/com/redhat/quarkus/snippets/quarkus-properties.json]: Added. I also added the default port for postgresql

File: microprofile.ls/com.redhat.microprofile.ls/src/main/resources/com/redhat/quarkus/snippets/quarkus-properties.json [hunk 6]
```diff
@@ -25,5 +25,19 @@
...
+		"body": [
+			"quarkus.container-image.group=$1",
+			"quarkus.container-image.registry=${2|docker.io,quay.io|}",
+			"quarkus.kubernetes.expose=${3|true,false|}",
+			"#quarkus.container-image.username=$4",
+			"#quarkus.container-image.password=$5"
+		],
+		"description": "Configure Container Image properties",
+		"context": {
+			"dependency": "quarkus-container-image"
```
Comment (fbricon) [inline microprofile.ls/com.redhat.microprofile.ls/src/main/resources/com/redhat/quarkus/snippets/quarkus-properties.json:40]: So this requirement is not sufficient. Users can only add quarkus-container-image-docker (not sure whether the container properties are relevant to the quarkus-container-image-jib and quarkus-container-image-s2i extensions). But the properties are not even proposed until quarkus-container-image-deployment.jar is added to the classpath (so here there's a discrepancy between the extension name and the jar)

Long story short, there's a missing link here to properly support container properties. This needs to be addressed in a new ticket.

In the meantime, I'd remove this snippet, as it's pretty much unusable until the above issue is fixed.

Comment (xorye) [reply microprofile.ls/com.redhat.microprofile.ls/src/main/resources/com/redhat/quarkus/snippets/quarkus-properties.json]: Thank you for looking into that, fixed
